### ISSUE TITLE
Create API endpoint => src/pages/api/projects/[id]/tasks.ts

### ISSUE THREAD

**Issue by kleneway** [PioneerSquareLabs/otto-playground#567]
Create the API endpoint src/pages/api/projects/[id]/tasks.ts. Here is the description: The tasks API route for the project detail page. Be sure to add the proper extension when creating the file. Here are the instructions:

## Summary

Create a new API endpoint `tasks.ts` for the project detail page. This endpoint will be responsible for fetching, creating, updating, and deleting tasks related to a specific project. The endpoint should support the following actions: GET, POST, PUT, and DELETE.

## Requirements

1. Create a new file named `tasks.ts` inside the `src/pages/api/projects/[id]` folder.

2. Import the necessary dependencies at the top of the file:

   ```typescript
   import { NextApiRequest, NextApiResponse } from "next";
   import { prisma } from "~/server/db";
   import { getServerAuthSession } from "~/server/auth";
   import { ProjectTask, projectTaskSchema } from "~/types";
   ```

3. Define a new async function named `handler` that takes `req` (NextApiRequest) and `res` (NextApiResponse) as parameters:

   ```typescript
   const handler = async (req: NextApiRequest, res: NextApiResponse) => {
     // Implementation goes here
   };
   ```

4. Inside the `handler` function, call the `getServerAuthSession` function to get the user's session:

   ```typescript
   const session = await getServerAuthSession({ req, res });
   const user = session?.user;
   ```

5. Check if the user is authenticated. If not, return a 401 Unauthorized status code with an error message:

   ```typescript
   if (!user) {
     return res.status(401).json({ error: "Unauthorized" });
   }
   ```

6. Extract the project ID from the request parameters:

   ```typescript
   const projectId = req.query.id as string;
   ```

7. Implement the GET action to fetch all tasks related to the specified project:

   - Check if the request method is GET:

     ```typescript
     if (req.method === "GET") {
       // Fetch tasks implementation
     }
     ```

   - Fetch the tasks from the database using Prisma:

     ```typescript
     const tasks = await prisma.projectTask.findMany({
       where: { projectId },
     });
     ```

   - Validate the fetched tasks using the Zod schema:

     ```typescript
     const validatedTasks = tasks.map((task) => projectTaskSchema.parse(task));
     ```

   - Return the validated tasks with a 200 OK status code:

     ```typescript
     return res.status(200).json(validatedTasks);
     ```

8. Implement the POST action to create a new task:

   - Check if the request method is POST:

     ```typescript
     if (req.method === "POST") {
       // Create task implementation
     }
     ```

   - Extract the task data from the request body and validate it using the Zod schema:

     ```typescript
     const taskData: ProjectTask = projectTaskSchema.parse(req.body);
     ```

   - Create the new task in the database using Prisma:

     ```typescript
     const newTask = await prisma.projectTask.create({
       data: { ...taskData, projectId },
     });
     ```

   - Return the created task with a 201 Created status code:

     ```typescript
     return res.status(201).json(newTask);
     ```

9. Implement the PUT action to update an existing task:

   - Check if the request method is PUT:

     ```typescript
     if (req.method === "PUT") {
       // Update task implementation
     }
     ```

   - Extract the task data from the request body and validate it using the Zod schema:

     ```typescript
     const taskData: ProjectTask = projectTaskSchema.parse(req.body);
     ```

   - Update the task in the database using Prisma:

     ```typescript
     const updatedTask = await prisma.projectTask.update({
       where: { id: taskData.id },
       data: taskData,
     });
     ```

   - Return the updated task with a 200 OK status code:

     ```typescript
     return res.status(200).json(updatedTask);
     ```

10. Implement the DELETE action to delete an existing task:

    - Check if the request method is DELETE:

      ```typescript
      if (req.method === "DELETE") {
        // Delete task implementation
      }
      ```

    - Extract the task ID from the request body:

      ```typescript
      const taskId = req.body.id;
      ```

    - Delete the task from the database using Prisma:

      ```typescript
      await prisma.projectTask.delete({ where: { id: taskId } });
      ```

    - Return a 204 No Content status code:

      ```typescript
      return res.status(204).end();
      ```

11. If none of the supported actions match the request method, return a 405 Method Not Allowed status code with an error message:

    ```typescript
    return res.status(405).json({ error: "Method Not Allowed" });
    ```

12. Export the `handler` function as the default export:

    ```typescript
    export default handler;
    ```

Once the implementation is complete, test the API endpoint using a tool like Postman or by integrating it with the frontend of the application. Make sure to test all supported actions (GET, POST, PUT, and DELETE) and verify that the tasks are fetched, created, updated, and deleted correctly.
